### PR TITLE
fix: multiple price rules margin.

### PR DESCRIPTION
**Pull request by dj12djdjs**
Multiple pricing rules are only applied to discount amount or percentage and disregards margin amount and percentage noted in the this issue #20791 
I have some further information at this [discussion.](https://discuss.erpnext.com/t/pricing-rule-w-apply-multiple-pricing-rules-margin/72118)

This PR applies the same logic to margin as already existed for discount rate.

### CODE REVIEW

File: erpnext/accounts/doctype/pricing_rule/pricing_rule.py [hunk 1]
```diff
@@ -330,13 +330,16 @@ def get_pricing_rule_details(args, pricing_rule):
...
-		item_details.margin_rate_or_amount = 0.0
+	for apply_on in ['Percentage', 'Amount']:
+		if pricing_rule.margin_type != apply_on: continue
+
+		field = 'margin_rate_or_amount'
+		if field not in item_details:
+                        item_details.setdefault(field, 0)
+                        item_details.setdefault('margin_type', apply_on)
+
+		item_details[field] += (pricing_rule.get(field, 0)
```
Comment (barredterra) [inline erpnext/accounts/doctype/pricing_rule/pricing_rule.py:342]: How can I understand the `+=`? Margins from multiple price rules get added together? This changes the previous behaviour, right?

**Review by barredterra — CHANGES_REQUESTED** [review]

File: erpnext/accounts/doctype/pricing_rule/pricing_rule.py (hunk 1 above)
Comment (dj12djdjs) [reply erpnext/accounts/doctype/pricing_rule/pricing_rule.py:342]: Yes, this does alter the previous behavior of pricing rules.

So, the new behavior is just copying what is being applied to discount from multiple price rules.
https://github.com/frappe/erpnext/blob/c6a1e3b450e3ce585da374fe41d041bae68cfcc5/erpnext/accounts/doctype/pricing_rule/pricing_rule.py#L356-L368

Previously, when multiple pricing rules are applied only the last pricing rules margin would be applied.
https://github.com/frappe/erpnext/blob/c6a1e3b450e3ce585da374fe41d041bae68cfcc5/erpnext/accounts/doctype/pricing_rule/pricing_rule.py#L333-L339

[Here](https://discuss.erpnext.com/t/pricing-rule-w-apply-multiple-pricing-rules-margin/72118) you can see the behavior previous to this patch in action vs what I was expecting.

Comment (barredterra) [reply erpnext/accounts/doctype/pricing_rule/pricing_rule.py:342]: Okay, makes sense. Thanks for clarifying.

**Comment by dj12djdjs** [comment]
Are we switching coding style to use spaces instead of tabs? I keep getting Sider problems that indentation is using tabs.

**Review by barredterra — COMMENTED** [review]
Sorry to request changes again, but maybe consider adding a test to ensure your desired behavior works and will continue to work in the future. As an example, see this test case: https://github.com/frappe/erpnext/blob/87dddbe868e0aa57bee141ae94986b6bbdd7abaf/erpnext/accounts/doctype/pricing_rule/test_pricing_rule.py#L103-L148

**Review by nabinhait — APPROVED** [review]

**Comment by nabinhait** [comment]
@dj12djdjs Can you please send another PR to the develop branch?

**Comment by dj12djdjs** [comment]
@nabinhait @barredterra 

Yes I will redo on develop with some tests.

**Pull request merged**